### PR TITLE
fix while loops

### LINKED ISSUE #1324: While loops push two values to the context var stack, but only pop one
[Try it Online!](https://vyxal.pythonanywhere.com/#WyJjIiwiIiwiYGFiY2RgKHtYfW7igrQiLCIiLCIiXQ==)

### PR DESCRIPTION
**Pull request by Steffan153**
Closes #1324

### CODE REVIEW

**Review by chunkybanana — APPROVED** [review]
Huh.

File: vyxal/transpile.py [hunk 1]
```diff
@@ -277,13 +277,16 @@ def transpile_structure(
             transpile_ast(struct.condition, indent, options=options)
             + indent_str("condition = pop(stack, 1, ctx=ctx)", indent)
             + indent_str("counter = ctx.range_start", indent)
+            + indent_str("temp_len = len(ctx.context_values)", indent)
```
Comment (chunkybanana) [inline vyxal/transpile.py:280]: Nested while loops may break this

**Review by chunkybanana — CHANGES_REQUESTED** [review]

**Comment by Lyxal** [comment]
Why not just pop the second context item it pushes? Or why not remove one of the context items being pushed?

**Comment by Steffan153** [comment]
> Why not just pop the second context item it pushes? Or why not remove one of the context items being pushed?

The condition being pushed was not needed, so I removed that. The rest of the code was because if `break` is used, then the context var is never popped.

File: vyxal/transpile.py (hunk 1 above)
Comment (Steffan153) [reply vyxal/transpile.py]: Good catch, fixed.

**Review by chunkybanana — APPROVED** [review]
LGTM

**Review by Lyxal — APPROVED** [review]

**Pull request merged**